plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php

```diff
@@ -304,46 +305,9 @@ public function onDisplay(
             }
         }
 
-        // Template
-        $templates = [];
-
-        if (!empty($allButtons['template'])) {
-            // Do we have a custom content_template_path
-            $template_path = $levelParams->get('content_template_path');
-            $template_path = $template_path ? '/templates/' . $template_path : '/media/vendor/tinymce/templates';
-
-            $filepaths = Folder::exists(JPATH_ROOT . $template_path)
-                ? Folder::files(JPATH_ROOT . $template_path, '\.(html|txt)$', false, true)
-                : [];
-
-            foreach ($filepaths as $filepath) {
-                $fileinfo      = pathinfo($filepath);
-                $filename      = $fileinfo['filename'];
-                $full_filename = $fileinfo['basename'];
-
-                if ($filename === 'index') {
-                    continue;
-                }
-
-                $title       = $filename;
-                $title_upper = strtoupper($filename);
-                $description = ' ';
-
-                if ($language->hasKey('PLG_TINY_TEMPLATE_' . $title_upper . '_TITLE')) {
-                    $title = Text::_('PLG_TINY_TEMPLATE_' . $title_upper . '_TITLE');
-                }
-
-                if ($language->hasKey('PLG_TINY_TEMPLATE_' . $title_upper . '_DESC')) {
-                    $description = Text::_('PLG_TINY_TEMPLATE_' . $title_upper . '_DESC');
-                }
-
-                $templates[] = [
-                    'title'       => $title,
-                    'description' => $description,
-                    'url'         => Uri::root(true) . $template_path . '/' . $full_filename,
-                ];
-            }
-        }
+        $jtemplates = !empty($allButtons['jtemplate'])
+            ? Uri::base(true) . '/index.php?option=com_ajax&plugin=tinymce&group=editors&format=json&format=json&template=' . $levelParams->get('content_template_path') . '&' . $csrf . '=1'
+            : false;
 
         // Check for extra plugins, from the setoptions form
         foreach (['wordcount' => 1, 'advlist' => 1, 'autosave' => 1, 'textpattern' => 0] as $pName => $def) {
```

Comment:
```suggestion
        foreach (['wordcount' => 1, 'advlist' => 1, 'autosave' => 1] as $pName => $def) {
```

textpattern is not an external plugin any more

Reply:
Thanks for researching this